src/main/java/ch/njol/skript/effects/EffExplodeWithFire.java

```diff
@@ -0,0 +1,93 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.effects;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.Kleenean;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Explosive;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.ExplosionPrimeEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Explode With Fire")
+@Description("Sets if an entity will explode with fire. Use \"make the explosion fiery\" in explosion prime events.")
+@Examples({"on explosion prime:", 
+	"\tmake the explosion fiery"})
+@Since("INSERT VERSION")
+public class EffExplodeWithFire extends Effect {
+
+	static {
+		Skript.registerEffect(EffExplodeWithFire.class,
+				"make %entities%'[s] explosion [(1¦not)] (fiery|incendiary)",
+				"make %entities% [(1¦not)] cause (a fiery|an incendiary explosion)",
+				"make the [event(-| )]explosion [(1¦not)] fiery");
```

Comment:
```suggestion
				"[(1¦de)]combust [the] (%-entities%|[event(-| )]explosion)",
				"make %entities% [(1¦not)] incendiary");
```
May have to add the null flag to the second pattern if it not having it causes issues.